misc/springsystem/src/processors/springsystemprocessor.cpp

```diff
@@ -0,0 +1,286 @@
+    springIndices->reserve(springs.size() * 2);
+
+    for (auto&& item : util::enumerate(positions, forces)) {
+        springMesh->addVertex(vec3(item.second(), 0.0f));
+        const auto pick = static_cast<uint32_t>(nodePicking_.getPickingId(item.first()));
+        const auto meta = static_cast<float>(glm::length(item.third()) / scaleFactor_.get());
+        const auto ind = nodeMesh->addVertex(vec3(item.second(), 0.0f), pick, meta);
+        nodeIndices->add(ind);
+    }
+    for (const auto& spring : springs) {
+        springIndices->add(static_cast<uint32_t>(spring.first));
+        springIndices->add(static_cast<uint32_t>(spring.second));
+    }
+
+    nodeOutport_.setData(nodeMesh);
+    springOutport_.setData(springMesh);
+}
+
+void SpringSystemProcessor::updateSystemFromProperties() { springSystem_ = getSystem(); }
+
+GravitySpringSystem<2, double> SpringSystemProcessor::getSystem() {
+    meshDirty_ = true;
+
+    auto grid = [&]() {
+        switch (springLayout_.get()) {
+            default:
+                [[fallthrough]];
+            case 1: {
+                return springmass::createLineGrid<2, double>(nodeCount_, nodeSpacing_);
+            }
+            case 2: {
+                const double offset = (std::ptrdiff_t{nodeCount_} - 1) * nodeSpacing_ * 0.5;
+                const dvec2 origin(-offset, offset);
+                return springmass::createRectangularGrid<2, double>(size2_t(nodeCount_), origin,
+                                                                    dvec2(nodeSpacing_));
+            }
+            case 3: {
+                const double offset = (nodeCount_ - 1) * nodeSpacing_ * 0.5;
+                const dvec2 origin(-offset, offset);
+                return springmass::createDiagonalGridDiagonal<2, double>(
+                    size2_t(nodeCount_), origin, dvec2(nodeSpacing_));
+            }
+            case 4: {
+                const double offset = (nodeCount_ - 1) * nodeSpacing_ * 0.5;
+                const dvec2 origin(-offset, offset);
+                return springmass::createHexagonalGrid<2, double>(size2_t(nodeCount_), origin,
+                                                                  dvec2(nodeSpacing_));
+            }
+        }
+    }();
+
+    return GravitySpringSystem<2, double>{deltaT_,
+                                          std::move(grid.positions),
+                                          std::move(grid.springs),
+                                          std::move(grid.locked),
+                                          externalForce_,
+                                          nodeMass_,
+                                          springConst_,
+                                          springRestLength_,
+                                          dampingCoeff_};
+}
+
+void SpringSystemProcessor::handlePicking(PickingEvent* p) {
+    if (p->getPressState() == PickingPressState::Move &&
```

Comment:
need to check whether picking state is `updated`. Otherwise you might handle events originating from other picked ids.

Reply:
How? Why? 
pickingpressstate::Move should imply update?